The first JVM-only extension

ppalaga opened the pull request:
Fix #894

ppalaga commented:
Any feedback is appreciated.

jamesnetherton commented on tooling/create-extension-templates/runtime-pom.xml:
```diff
@@ -30,9 +30,10 @@
 
     <properties>
-        <firstVersion>1.0.0-M6</firstVersion>
+        <firstVersion>1.0.0-M5</firstVersion>
```
Shouldn't it be M6?

ppalaga replied:
+1

jamesnetherton commented on tooling/create-extension-templates/quarkus-extension.yaml:
```diff
@@ -0,0 +1,35 @@
+  - "[=category]"
+[/#list]
+[/#if]
+  status: "[=nativeSupported?then('stable', 'preview')]"
```
Auto generated `quarkus-extension.yaml` :heart_eyes:

ppalaga commented:
There are two things missing:

* Running the JVM-only tests by the CI
* Listing the JVM-only extensions in the docs

ppalaga commented:
> There are two things missing:
> 
> * Running the JVM-only tests by the CI

It should already be covered by the `build` jobs's `mvn clean install`.

> * Listing the JVM-only extensions in the docs

Can be done as a followup.

ppalaga commented:
ec0ec5f:

* Fixed the M5/6 issue found by @jamesnetherton 
* Replaced the plugin in the docs
* Used the proper release of the plugin to generate the extension

ppalaga commented:
Another followup: some source checks may need a reconfig

ppalaga commented:
@jamesnetherton reviewed and agreed to merge
